### PR TITLE
[2.16.x] DDF-5311 Added support for upgrading some security configs with migration

### PR DESCRIPTION
**Pull request by tyler30clemens**
### What does this PR do? 
DO NOT SQUASH!!! 

Adds support for upgrading `Web Context Policy Manger`, `STS Guest Claims Handler`, `STS Server`, and `Session` security configurations with migration across versions.

#### Who is reviewing it? 
@austinsteffes 
@ahoffer 

#### Select relevant component teams: 
@codice/core-apis 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter
@paouelle 

#### How should this be tested?
1. Install an old release version of DDF that is in the list of "supported versions" and create `Web Context Policy Manger`, `STS Guest Claims Handler`, `STS Server`, and `Session` configurations under the Security App
2. Run migration:export, which creates 3 files under the `etc/exported` directory
3. Shutdown
4. Start up a DDF built on this PR's changes
5. Install with `profile:install standard`
6. Copy all 3 files from step 2 into this newly installed DDF under /exported.
7. Run `migration:import -f`
8. After the system automatically restarts, verify all config changes are present in the new DDF and they work

#### What are the relevant tickets?
Work for: #5311 

#### Checklist:
- [X] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/resources/content/_managing/_configuring/reusing-configurations.adoc [hunk 1]
```diff
@@ -76,5 +76,12 @@ The same step-by-step process above can be followed when migrating configuration
 ** Keystores
 ** Truststores
 ** Service wrapper `*.conf` files, if the ${branding} is installed as a service
-** Content Directory Monitor Configuration (including URL Resource Reader)
+** Select AdminUI Configurations, Including:
+*** Content Directory Monitor Configuration (including URL Resource Reader)
```
Comment (brjeter) [inline distribution/docs/src/main/resources/content/_managing/_configuring/reusing-configurations.adoc]: ✏️ I think we should put URL Resource Reader on its own line now.

File: distribution/docs/src/main/resources/content/_managing/_configuring/reusing-configurations.adoc [hunk 2]
```diff
@@ -76,5 +76,12 @@ The same step-by-step process above can be followed when migrating configuration
...
+*** Web Context Policy Manager
+*** Security STS Guest Claims Handler
+*** Security STS Server
+*** Session
+[WARNING]
+If a supported configuration is being imported across versions, any corresponding `.config` files in the etc directory will not be put into the etc directory of the importing system.
```
Comment (brjeter) [inline distribution/docs/src/main/resources/content/_managing/_configuring/reusing-configurations.adoc]: ❓ How does this render with the warning message inside the bullet point list?

File: platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java [hunk 3]
```diff
@@ -46,11 +48,25 @@
...
   private static final String CURRENT_VERSION = "1.0";
 
-  private static final String CONTENT_DIRECTORY_MONITOR_FPID =
-      "org.codice.ddf.catalog.content.monitor.ContentDirectoryMonitor";
-
-  private static final String URL_RESOURCE_READER_PID =
-      "ddf.catalog.resource.impl.URLResourceReader";
+  private static final List<String> ACCEPTED_ENTRY_PIDS =
+      Collections.unmodifiableList(
+          new ArrayList<String>() {
```
Comment (brjeter) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java]: ✏️ `Arrays.asList()` would be a bit cleaner
```
private static final List<String> ACCEPTED_ENTRY_PIDS =
      Arrays.asList(
          "ddf.catalog.resource.impl.URLResourceReader",
          "org.codice.ddf.security.policy.context.impl.PolicyManager",
          "ddf.security.sts.guestclaims",
          "ddf.security.sts",
          "ddf.security.http.impl.HttpSessionFactory");
```

File: platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java [hunk 4]
```diff
@@ -46,11 +48,25 @@
...
+            }
+          });
+
+  private static final List<String> ACCEPTED_ENTRY_FACTORY_PIDS =
+      Collections.unmodifiableList(
+          new ArrayList<String>() {
+            {
+              add("org.codice.ddf.catalog.content.monitor.ContentDirectoryMonitor");
+            }
+          });
```
Comment (brjeter) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java]: ✏️ Should use `Collections.singletonList` or `Arrays.asList()`
```
  private static final List<String> ACCEPTED_ENTRY_FACTORY_PIDS =
      Collections.singletonList("org.codice.ddf.catalog.content.monitor.ContentDirectoryMonitor");
```

File: platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java [hunk 5]
```diff
@@ -134,13 +150,29 @@ public void doVersionUpgradeImport(ImportMigrationContext context) {
...
-                CONTENT_DIRECTORY_MONITOR_FPID.equals(s.getFactoryPid())
-                    || s.getPid().equals(URL_RESOURCE_READER_PID))
+        .filter(s -> filterConfigurationAdminEntries(s))
         .forEach(ImportMigrationConfigurationAdminEntry::restore);
   }
 
+  private boolean filterConfigurationAdminEntries(ImportMigrationConfigurationAdminEntry entry) {
+    if (entry.getPid().equals("org.codice.ddf.security.filter.login.Session")) {
+      entry.setPid("ddf.security.http.impl.HttpSessionFactory");
+    }
```
Comment (brjeter) [inline platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigratable.java]: ✏️ We should do this "conversion" logic separately and after the filtering by pid/factoryPid.

File: distribution/docs/src/main/resources/content/_managing/_configuring/reusing-configurations.adoc (hunk 2 above)
Comment (tyler30clemens) [reply distribution/docs/src/main/resources/content/_managing/_configuring/reusing-configurations.adoc]: image

Would it be better to put it under that last bullet point, or is this good as is? (after putting URL Resource Reader on its own line)

Comment (brjeter) [reply distribution/docs/src/main/resources/content/_managing/_configuring/reusing-configurations.adoc]: I think that's fine 👍

**Comment by brjeter** [comment]
build now

**Comment by brjeter** [comment]
Hero successful ✅ Tested on a DDF 2.13.3 and 2.16.0.

**Comment by tyler30clemens** [comment]
rebased 
build now

**Comment by tyler30clemens** [comment]
build now

**Comment by brjeter** [comment]
build now

**Comment by tyler30clemens** [comment]
updated suppressions
build now

**Comment by brjeter** [comment]
Merging without successful CI since the last one got through to OWASP and that has been fixed and passed locally.

**Pull request merged**